zxlive/vitem.py

```diff
@@ -256,6 +301,12 @@ def mouseReleaseEvent(self, e: QGraphicsSceneMouseEvent) -> None:
             return
         if e.button() == Qt.MouseButton.LeftButton:
             if self._old_pos != self.pos():
+                if self.g.type(self.v) == VertexType.W_INPUT:
+                    # set the position of w_in to next to w_out at the same angle
+                    w_out = get_w_partner_vitem(self.g, self.graph_scene, self.v)
+                    w_in_pos = w_out.pos() + QPointF(0, 0.3 *SCALE)
+                    w_in_pos = rotate_point(w_in_pos, w_out.pos(), w_out.rotation())
+                    self.setPos(w_in_pos)
```

Comment:
Don't we also have to update the position in the graph?

Reply:
This is done inderectly in the lines 315 - 318, where the vertices_moved signal is emitted
```python
else:
    scene.vertices_moved.emit([
        (it.v, *pos_from_view(it.pos().x(),it.pos().y()))
        for it in scene.selectedItems() if isinstance(it, VItem)
    ])
```

Reply:
This code actually needed some modification to also update the w partner in the graph. Fixed in https://github.com/Quantomatic/zxlive/pull/77/commits/8dd6533acdd12690c9c67a04ed9c62d3ad238334

Reply:
That makes sense 👍